test/functional/legacy/013_autocommands_on_close_spec.lua

```diff
@@ -0,0 +1,108 @@
+-- Tests for autocommands on :close command
+-- Write three files and open them, each in a window.
+-- Then go to next window, with autocommand that deletes the previous one.
+-- Do this twice, writing the file.
+-- Also test deleting the buffer on a Unload event.  If this goes wrong there
+-- will be the ATTENTION prompt.
+-- Also test changing buffers in a BufDel autocommand.  If this goes wrong there
+-- are ml_line errors and/or a Crash.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('autocommands_on_close', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([=[
+      start of testfile
+      	contents
+      	contents
+      	contents
+      end of testfile]=])
+
+    -- Write the current buffer to "test13.in", else subsequent
+    -- steps will fail
+    execute('w! test13.in')
+    execute('?^start of testfile?,?^end of testfile?w! Xtestje1')
+    execute('?^start of testfile?,?^end of testfile?w! Xtestje2')
+    execute('?^start of testfile?,?^end of testfile?w! Xtestje3')
```

Comment:
If you look inside `helpers.execute` you can see that strings beginning with `/` are special-cased and don't get the `:` to make them into an ex command. I think this is meant for convenience because many tests use forward search to position the cursor.

Anyway you can make it work by simply prepending the colon yourself:

    execute(':/^start of testfile/,/^end of testfile/w! Xtestje1')